pkg/applicationcontroller/applicationinformers/controller_test.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2022 Authors of spidernet-io
+// SPDX-License-Identifier: Apache-2.0
+
+package applicationinformers
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/spidernet-io/spiderpool/pkg/logutils"
+)
+
+var _ = Describe("Controller", Label("unitest"), func() {
+	Context("NewApplicationController", func() {
+		log := logutils.Logger.Named("application-informers-ut")
+
+		It("fail to new application controller with no reconcile function", func() {
+			_, err := NewApplicationController(nil, fakeCleanupFunc, log)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("fail to new application controller with no cleanup function", func() {
+			_, err := NewApplicationController(fakeReconcileFunc, nil, log)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("new application controller successfully", func() {
+			controller, err := NewApplicationController(fakeReconcileFunc, fakeCleanupFunc, log)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(controller).NotTo(BeNil())
+		})
+	})
+})
```

Comment:
The code looks like unit test code for a function called `NewApplicationController` in the `applicationinformers` package. It tests three different scenarios, including both failure and success cases. Overall, the style of testing appears to follow the standard ginkgo framework.

Since this is a unit test, there should be no bug risks associated with this code. However, one small improvement suggestion is that it would be helpful to see the definition or declaration of `fakeReconcileFunc` and `fakeCleanupFunc`, as these functions would determine the specific conditions being tested.